.github/workflows/check_python_dependencies.yml

```
@@ -0,0 +1,50 @@
+name: Check Python Dependencies
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  check_dependencies:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Clone repo-tools repo
+        uses: actions/checkout@v4
+        with:
+          repository: salman2013/repo-tools
+          path: repo_tools
+          ref: salman/add-script-for-python-dependencies
+                    
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Install setuptool
+        run: pip install setuptools
+      
+      - name: Install requests module
+        run: pip install requests
+
+      - name: Install rich module
+        run: pip install rich
+
+      - name: Install requirements parser module
+        run: pip install requirements-parser  
+
+      - name: Make the script files executable
+        run: chmod +x repo_tools/edx_repo_tools/find_dependencies/find_python_dependencies.py
+
+      - name: Run Python script
+        run: python repo_tools/edx_repo_tools/find_dependencies/find_python_dependencies.py requirements/edx/base.txt requirements/edx/testing.txt 
```

```suggestion
        # Tracking ignored repos here: https://github.com/openedx/public-engineering/issues/174
        run: python repo_tools/edx_repo_tools/find_dependencies/find_python_dependencies.py requirements/edx/base.txt requirements/edx/testing.txt   --ignore https://github.com/edx/codejail-includes --ignore https://github.com/edx/braze-client --ignore https://github.com/edx/edx-name-affirmation --ignore https://github.com/mitodl/edx-sga --ignore https://github.com/edx/token-utils --ignore https://github.com/edx/codejail-includes --ignore https://github.com/edx/braze-client --ignore https://github.com/edx/edx-name-affirmation --ignore https://github.com/mitodl/edx-sga --ignore https://github.com/edx/token-utils
```

I've made sure that all these repos are part of a tickte to further investigate and clean them up here: https://github.com/openedx/public-engineering/issues/174 so we can ignore them for now and prevent future additions, we'll make sure to remove the repos from this list as they get removed from edx-platform.